### PR TITLE
[SYCL] Matching sizeof long double of SYCL device with that of host

### PR DESCRIPTION
**Pull request by schittir**
static_assert of sizeof(long double)>8 fails to compile with -fsycl
switch because sizeof(long double) for SYCL device is 8 bytes whereas
the host is 16 bytes. This commit sets sizeof(long double) width to 16
bytes to match that of SYCL host for Linux gnu SYCLdevice x86_64 and x86

### CODE REVIEW

File: clang/lib/Basic/Targets/SPIR.h [hunk 1]
```diff
@@ -273,6 +273,29 @@ class LLVM_LIBRARY_VISIBILITY MicrosoftX86_64_SPIR64TargetInfo
...
+};
+
+// x86-32 SPIR Linux target
+class LLVM_LIBRARY_VISIBILITY LinuxX86_32_SPIRTargetInfo
+    : public LinuxTargetInfo<SPIR32SYCLDeviceTargetInfo> {
+public:
+  LinuxX86_32_SPIRTargetInfo(const llvm::Triple &Triple,
+                             const TargetOptions &Opts)
+      : LinuxTargetInfo<SPIR32SYCLDeviceTargetInfo>(Triple, Opts) {
+    LongDoubleWidth = LongDoubleAlign = 128;
```
Comment (erichkeane) [inline clang/lib/Basic/Targets/SPIR.h:295]: This doesn't seem right, is it?  Linux 32 bit is going to have its long double width modified a bunch, isn't it?  Theres a bunch of flags to change this size.

**Comment by Fznamznon** [comment]
> @Fznamznon : That one is interesting! I wonder if we should just do that for SYCL as well (modify that 'if' statement). I think this one is 'the right' way, but if community has already made that selection perhaps we need to as well.

I think we should be aligned with community code and generalize approaches between SYCL and other offloading models which already is presented in master (like OpenMP), because they solve similar problems quite often. One problem I know with this community method - it crashes if user didn't specify AUX triple, but It is easy to fix by adding check on AUX triple presence.
If the method which this PR exposes is better (I cannot say for sure, I'm not an expert in this place), I think we should generalize approach between OpenMP and SYCL and upstream it.

File: clang/lib/Basic/Targets/SPIR.h [hunk 2]
```diff
@@ -273,6 +269,7 @@ class LLVM_LIBRARY_VISIBILITY MicrosoftX86_64_SPIR64TargetInfo
     Builder.defineMacro("_M_AMD64", "100");
   }
 };
+
```
Comment (erichkeane) [inline clang/lib/Basic/Targets/SPIR.h:272]: unrelated change.

File: clang/test/SemaSYCL/check-long-double-size.cpp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -aux-triple x86_64-unknown-linux-gnu -fsyntax-only -DLONG_DOUBLE_SIZE=16 -verify %s
+// expected-no-diagnostics
```
Comment (erichkeane) [inline clang/test/SemaSYCL/check-long-double-size.cpp:2]: Only 1 of these expected-no-diagnostics should exist.
